From eb32eb8c96d3948823b8467dd1837aa3d3c5ec27 Mon Sep 17 00:00:00 2001
From: Jon TURNEY <jon.turney@dronecode.org.uk>
Date: Sat, 12 Jul 2014 13:33:47 +0100
Subject: [PATCH 02/24] Avoid O_BINARY redefinition warning on Cygwin

Avoid O_BINARY redefinition warning compiling processor for Cygwin

On Cygwin, WIN32 not defined, but O_BINARY is defined by fcntl.h, so avoid a
redefinition warning.

../src/processor/minidump.cc:55:0: warning: "O_BINARY" redefined [enabled by default]
 #define O_BINARY 0
 ^
In file included from /usr/include/sys/fcntl.h:3:0,
                 from /usr/include/fcntl.h:14,
                 from ../src/processor/minidump.cc:39:
/usr/include/sys/_default_fcntl.h:52:0: note: this is the location of the previous definition
 #define O_BINARY _FBINARY
 ^

Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
---
 src/processor/minidump.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/processor/minidump.cc b/src/processor/minidump.cc
index 8ae260d..a4bf55a 100644
--- a/src/processor/minidump.cc
+++ b/src/processor/minidump.cc
@@ -49,7 +49,9 @@
 #define snprintf _snprintf
 #else  // _WIN32
 #include <unistd.h>
+#ifndef O_BINARY
 #define O_BINARY 0
+#endif
 #endif  // _WIN32
 
 #include <fstream>
-- 
2.1.1

